src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationAndGeneratorDriverTranslationAction.cs

```diff
@@ -15,6 +15,18 @@ public virtual Task<Compilation> TransformCompilationAsync(Compilation oldCompil
             {
                 return Task.FromResult(oldCompilation);
             }
+
+            /// <summary>
+            /// Whether or not <see cref="TransformCompilationAsync" /> can be called on Compilations that may contain
+            /// generated documents.
+            /// </summary>
+            /// <remarks>
+            /// Most translation actions add or remove a single syntax tree which means we can do the "same" change
+            /// to a compilation that contains the generated files and one that doesn't; however some translation actions
+            /// (like <see cref="ReplaceAllSyntaxTreesAction"/>) will unilaterally remove all trees, and would have unexpected
+            /// side effects. This opts those out of operating on ones with generated documents where there would be side effects.
+            /// </remarks>
+            public virtual bool CanApplyToCompilationWithGeneratedDocuments => false;
```

Comment:
i'd prefer this be abstract and have all subclasses be forced to consider what the right answer here is.  that said, if we do make this virtual, i approve of it be conservative by default.

Reply:
Done.